### PR TITLE
Clear Package Folder Before Generation

### PR DESCRIPTION
**Pull request by wanlwanl**
Fix: https://github.com/Azure/azure-sdk-tools/issues/8057
Test pipeline: https://github.com/test-repo-billy/azure-rest-api-specs/pull/3473

### CODE REVIEW

File: tools/js-sdk-release-tools/src/common/utils.ts [hunk 1]
```diff
@@ -73,4 +93,20 @@ export function tryReadNpmPackageChangelog(packageFolderPath: string): string {
...
+    const tspConfig = await loadTspConfig(typeSpecDirectory);
+    const serviceDir = tspConfig.parameters?.['service-dir']?.default;
+    if (!serviceDir) {
+        throw new Error(`Misses service-dir in parameters section of tspconfig.yaml. ${messageToTspConfigSample}`);
+    }
+    const packageDir = tspConfig.options?.[emitterName]?.['package-dir'];
+    if (!packageDir) {
+        throw new Error(`Misses package-dir in ${emitterName} options of tspconfig.yaml. ${messageToTspConfigSample}`);
+    }
+    const packageDirFromRoot = posix.join(serviceDir, packageDir);
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/common/utils.ts:110]: just curious why posix join? what if this command runs in windows environment?

**Review by MaryGao — APPROVED** [review]
Left small comments!

Regarding the customization code I remember Ray has mentioned we could improve our automation to have message like `The SDK has customized SDK code and might not be compatible with the new specs, the service team may need to update the customized code or the new specs`. Not sure this is feasible and we have plan for this.

File: tools/js-sdk-release-tools/src/common/utils.ts (hunk 1 above)
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/common/utils.ts:110]: it's used in config/package that only accept posix format path.

**Comment by wanlwanl** [comment]
> Left small comments!
> 
> Regarding the customization code I remember Ray has mentioned we could improve our automation to have message like `The SDK has customized SDK code and might not be compatible with the new specs, the service team may need to update the customized code or the new specs`. Not sure this is feasible and we have plan for this.

The current customized code is mixed in generated code, we don't know if customer has customized code. I think it's better to seperate generated code and customized code in the future.

**Pull request merged**